projects/vdk-core/src/vdk/internal/builtin_plugins/connection/database_managed_connection.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2023-2024 Broadcom
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Container
+from typing import Optional
+
+from vdk.internal.builtin_plugins.connection.decoration_cursor import DecorationCursor
+from vdk.internal.builtin_plugins.connection.execution_cursor import ExecuteOperationResult
+from vdk.internal.builtin_plugins.connection.execution_cursor import ExecutionCursor
+from vdk.internal.builtin_plugins.connection.recovery_cursor import RecoveryCursor
+
+
+class IDatabaseManagedConnection:
+    def db_connection_validate_operation(
+        self, operation: str, parameters: Optional[Container]
+    ) -> None:
+        """
+        Validates the query and parameters.
+
+        For example:
+        _db_connection_validate_operation(operation, parameters):
+            if(_max_query_length_limit_exceeded(operation, parameters)):
+                raise Exception("max query length limit exceeded")
+
+        :param operation: str
+            Database operation - a SQL query or command to be executed.
+        :param parameters: Optional[Container]
+            Parameters may be provided as sequence or mapping and will be bound to variables in the operation.
+            Variables are specified in a database-specific notation. See chosen database documentation for details.
+        :return:
+        """
+        pass
+
+    def db_connection_decorate_operation(
+        self, decoration_cursor: DecorationCursor
+    ) -> None:
+        """
+        Curates the operation and parameters.
+        If an exception is raised, the cursor execution will be suspended.
+
+        For example:
+        _db_connection_decorate_operation(decoration_cursor: DecorationCursor):
+            managed_operation = decoration_cursor.get_managed_operation()
+            managed_operation.set_operation("prefix" +
+                                            managed_operation.get_operation())
+
+        :param decoration_cursor: DecorationCursor
+            A PEP249Cursor implementation purposed for query and parameters decoration.
+            Provides operation details and tooling.
+        :return:
+        """
+        pass
+
+    def db_connection_execute_operation(
```

Comment:
execute_operation should have a default implementation. I think we can put it here, or override it in ManagedConnectionBase, so that this class remains an "interface".

Reply:
@DeltaMichael  won't it cause problems with the checks if we override it in ManagedConnectionBase? I would probably need to change the checks to check whether it is ManagedConnectionBase which is ok but rn with this implementation we will never end up using the default of the interface we will always end up using the default of the hooks

Reply:
It doesn't matter right now. However, we will need a default implementation when we start migrating the database plugins, e.g. vdk-oracle will need a default `execute_operation` method, even if it doesn't override it.

Reply:
I will add it now